moor/lib/src/runtime/types/custom_type.dart

```diff
@@ -39,3 +39,30 @@ class EnumIndexConverter<T> extends TypeConverter<T, int> {
     return (value as dynamic)?.index as int;
   }
 }
+
+/// Enforces nnbd for TypeConverters
```

Comment:
I think this is a little generic, how about something like

```suggestion
/// A type converter automatically mapping `null` values to `null` in both directions.
///
/// Instead of overriding  [mapToDart] and [mapToSql], subclasses of this converter
/// should implement [requireMapToDart] and [requireMapToSql], which are used to
/// map non-null values to and from sql values, respectively.
///
/// Apart from the implementation changes, subclasses of this converter can be used
/// just like all other type converters.
```

(you may have to wrap some lines differently)